x-pack/plugins/cases/public/common/mock/test_providers.tsx

```diff
@@ -9,31 +9,41 @@ import { euiDarkVars } from '@kbn/ui-shared-deps-src/theme';
 import { I18nProvider } from '@kbn/i18n-react';
 import React from 'react';
 import { ThemeProvider } from 'styled-components';
-import { SECURITY_SOLUTION_OWNER } from '../../../common';
+import { CasesContextValue, DEFAULT_FEATURES, SECURITY_SOLUTION_OWNER } from '../../../common';
 import { CasesProvider } from '../../components/cases_context';
 import { createKibanaContextProviderMock } from '../lib/kibana/kibana_react.mock';
 import { FieldHook } from '../shared_imports';
 
 interface Props {
   children: React.ReactNode;
   userCanCrud?: boolean;
+  features?: CasesContextValue['features'];
 }
 
 window.scrollTo = jest.fn();
 const MockKibanaContextProvider = createKibanaContextProviderMock();
 
 /** A utility for wrapping children in the providers required to run most tests */
-const TestProvidersComponent: React.FC<Props> = ({ children, userCanCrud = true }) => (
-  <I18nProvider>
-    <MockKibanaContextProvider>
-      <ThemeProvider theme={() => ({ eui: euiDarkVars, darkMode: true })}>
-        <CasesProvider value={{ owner: [SECURITY_SOLUTION_OWNER], userCanCrud }}>
-          {children}
-        </CasesProvider>
-      </ThemeProvider>
-    </MockKibanaContextProvider>
-  </I18nProvider>
-);
+const TestProvidersComponent: React.FC<Props> = ({
+  children,
+  userCanCrud = true,
+  features = {},
+}) => {
+  const featuresOptions = { ...DEFAULT_FEATURES, ...features };
```

Comment:
I think this spread overriding is a little bit too direct, maybe we can turn this into a merge operation and support partial overrides. 
For example, considering someday we add the _alert.foo_ feature with a default:
```
DEFAULT_FEATURES = {
  alerts: {
    sync: true,
    foo: true,
  },
};
```
The features prop override will always have to contain _foo_ along with _sync_, and we'll have to update every plugin passing some alerts features, otherwise, keeping the current `{ alerts: { sync: false } }`, with the current implementation, we will override the entire `alerts` entry and _alerts.foo_ will become undefined.
Maybe we can consider using [lodash `mergeWith`](https://lodash.com/docs/4.17.15#mergeWith)? so we can support partial feature overrides

Reply:
Yeah I agree having a `merge` will probably be helpful here instead of just a spread.